specifyweb/stored_queries/queryfieldspec.py

```diff
@@ -118,8 +121,7 @@ def from_stringid(cls, stringid, is_relation):
                 tree_rank = tree_id_match.group(1)
                 tree_field = 'ID'
             else:
-                tree_field_match = TAXON_FIELD_RE.match(extracted_fieldname) \
-                                   if node is datamodel.get_table('Taxon') else None
+                tree_field_match = TAXON_FIELD_RE.match(extracted_fieldname) if node is datamodel.get_table('Taxon') else GEOGRAPHY_FIELD_RE.match(extracted_fieldname) if node is datamodel.get_table('Geography') else None
```

Comment:
(optional) 
I can envision this line getting pretty long (at maximum a clause for every tree). 

If you want to keep readability, can you extract this into a separate function or object (dictionary) which can resolve a Table (and/or field name depending on how you want to approach this) to the correct regular expression matches? 

The end result would be something like: 
```py
tree_field_match = something
```
instead of the long conditional value we have currently.

Reply:
I think it'll be better to remove the restriction in the first place?

Reply:
That's ultimately the next step. I was drafting a PR locally that does exactly that, but I haven't been satisfied with the implementation yet. Maybe I'll revisit this in the coming weeks. If another dev wants to do this, feel free! Just LMK.

Reply:
@grantfitzsimmons excited to see your implementation! it might be easier for you to base it off https://github.com/specify/specify7/tree/wb_improvements, since it parses out tree ranks more nicely (instead of having the concept of tree_rank and tree_field, which makes things trivial things harder -- like if queryfield ends with a relationship or not), it makes a phantom query-builder relationship, making things much more simpler. 

The implementation I've in there also supports relatonships from rank (so you can do determination -> Species -> created by) which is not possible in the production implementation (not your fault, legacy implementation). The biggest blocker, as with most things, is a frontend implementation. I didn't touch it yet since geo guys will change relevant portions....